src/Jackalope/Transport/DoctrineDBAL/Client.php

```diff
@@ -460,14 +460,17 @@ public function setCaseSensitiveEncoding($encoding)
      */
     private function getCaseSensitiveEncoding()
     {
-        if (!$this->caseSensitiveEncoding) {
-            $params = $this->conn->getParams();
-            $charset = isset($params['charset']) ? $params['charset'] : 'utf8';
+        if ($this->caseSensitiveEncoding) {
```

Comment:
should we init this field with `null` and check `null !===` to cache the result? otherwise if we are on a non-case-sensitive encoding, we re-calculate the thing again and again for no value.